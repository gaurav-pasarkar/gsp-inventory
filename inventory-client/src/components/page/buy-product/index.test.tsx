import matchSnapshot from "../../../test-utils/match-snapshot";
import BuyProduct from "./index";
import {fireEvent, render, waitFor} from "@testing-library/react";

describe('Buy product', () => {

  it('should render', () => {
    matchSnapshot(<BuyProduct/>);
  });

  it('should render required field validations', async () => {
    const { getByText } = render(<BuyProduct/>);

    fireEvent.click(getByText(/Add product/));

    await waitFor(() => {
      expect(getByText(/Product name required/)).toBeInTheDocument();
      expect(getByText(/Cost price\(per unit\) required/)).toBeInTheDocument();
      expect(getByText(/Selling price\(per unit\) required/)).toBeInTheDocument();
      expect(getByText(/Quantity required/)).toBeInTheDocument();
    })
  })
});