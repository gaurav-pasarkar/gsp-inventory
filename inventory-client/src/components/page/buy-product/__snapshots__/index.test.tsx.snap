// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buy product should render 1`] = `
<form
  className="makeStyles-root-1"
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
    >
      Product Name
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-5"
        >
          <span>
            Product Name
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
    >
      Cost Price(per unit)
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="number"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-5"
        >
          <span>
            Cost Price(per unit)
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
    >
      Selling Price(per unit)
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="number"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-5"
        >
          <span>
            Selling Price(per unit)
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
    >
      Quantity
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="number"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-5"
        >
          <span>
            Quantity
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="makeStyles-cta-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Add product
      </span>
    </button>
  </div>
</form>
`;
